MatiasDure/Project_IMTS
# Speed up fish

**Description**
- Interactable fish
- Swims faster
- Bubble particles spawn (not very heavy - can tone down performance if needed)

**Showcase**
![FishTapping](https://github.com/user-attachments/assets/c69defe8-aa23-44b9-bbfa-904e576c3bc6)

**How to test**
- Go to scene SpeedUpFish
- Play
- Tap fish
- Fish should lerp it's speed to target speed, keep target speed for SpeedUpDuration seconds and then lerp it's speed back down to original speed. (SpeedUpDuration = Amount of time fish has target speed effect. LerpSpeedDuration = Amount of time it takes to reach / lerp to target speed - also used to lerp back down to original speed. SpeedUpDuration does not include LerpSpeedDuration, so the whole effect is SpeedUpDuration + (LerpSpeedDuration * 2)).
- Temporary bubble particles spawn from fish when is under the effect.
- Bubbles should stop after effect wears out.
- You can see the speed changing in the Inspector if you click on the Fish that is a child of otherWorld in Hierarchy
